lib/src/style/style.dart

```diff
@@ -40,9 +40,24 @@ class StyleReader {
   final Logger logger;
   final Map<String, String>? httpHeaders;
 
-  StyleReader(
-      {required this.uri, this.apiKey, Logger? logger, this.httpHeaders})
-      : logger = logger ?? const Logger.noop();
+  /// Callback to construct a vector tile provider given parameters read from
+  /// the style
+  ///
+  /// Defaults to a standard [VectorTileProvider].
+  final VectorTileProvider Function(
```

Comment:
This seems like a great improvement.
Can we extract a typedef? That will make it more readable.

Also let's rename `constructTileProvider` to `tileProviderFactory`